api/notd/token_metadata_processor.py

```diff
@@ -123,16 +132,30 @@ async def retrieve_token_metadata(self, registryAddress: str, tokenId: str) -> R
         if registryAddress in ('0x772Da237fc93ded712E5823b497Db5991CC6951e', '0xE072AA2B0d39587A08813391e84495971A098084'):
             # TODO(krishan711): Implement special case for Everdragons, DISTXR (their contract is unverified)
             raise TokenDoesNotExistException()
-        try:
-            tokenMetadataUriResponse = await self.ethClient.call_function(toAddress=registryAddress, contractAbi=self.erc721MetadataContractAbi, functionAbi=self.erc721MetadataUriFunctionAbi, arguments={'tokenId': int(tokenId)})
-        except BadRequestException as exception:
-            if 'URI query for nonexistent token' in exception.message:
-                raise TokenDoesNotExistException()
-            if 'execution reverted' in exception.message:
-                raise TokenDoesNotExistException()
-            if 'out of gas' in exception.message:
-                raise TokenDoesNotExistException()
-            raise exception
+        tokenMetadataUriResponse = []
+        if (await self.ethClient.call_function(toAddress=registryAddress, contractAbi=self.erc165MetadataContractAbi, functionAbi=self.erc165SupportInterfaceUriFunctionAbi, arguments={'interfaceId': '0x5b5e139f'}))[0]:
```

Comment:
extract the interface ids to the top of the file like 
```
_INTERFACE_ID_ERC721 = '0x5b5e139f'
```

Reply:
Also, please make a var for this calculationg and dont do directly in the if condition i.e.
```
isContractSupportingErc721 = (await ...)[0]
if (isContractSupportingErc721) {

}
```